### PR TITLE
Politician details page

### PR DESCRIPTION
**Pull request by AriHrannar**
## Description
Details page for politicians. Had to fix some flow stuff as well since we have normalized the data.

List of all politicians
politicianslist

Politicians detail page
politiciandetails

### CODE REVIEW

File: src/Components/Politicians/Politicians/index.js [hunk 1]
```diff
@@ -31,16 +31,20 @@ export class Politicians extends React.Component {
 
   render() {
     const { politicians } = this.props;
+    const politicianContainerStyles = {
+      display: 'flex',
```
Comment (Ragnar-H) [inline src/Components/Politicians/Politicians/index.js]: ❤️

File: src/Components/Politicians/Politicians/index.js [hunk 2]
```diff
@@ -31,16 +31,20 @@ export class Politicians extends React.Component {
...
+      justifyContent: 'space-around',
+      flexWrap: 'wrap',
+    };
     return (
-      <div>
+      <div style={politicianContainerStyles}>
```
Comment (Ragnar-H) [inline src/Components/Politicians/Politicians/index.js]: Nitpick:
I'd be happy to call this style `container`

But not important

File: src/Components/Politicians/PoliticianCard/index.js [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
   name: string,
   party: string,
-  progress: number,
 };
 
 const PoliticianCard = (props: Props) => {
-  const { name, party, progress } = props;
-  const cardStyle = {};
+  const { name } = props;
+  const cardStyle = { flex: '1 1 0', margin: '1%' };
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: Nitpick:

I guess you could define this (and any static styles) outside of the component.
Either in a separate file or just outside of the `PoliticianCard` function

File: src/Components/Politicians/PoliticianCard/index.js [hunk 4]
```diff
@@ -1,18 +1,35 @@
...
+  const { politician } = props;
+  const containerStyle = {
+    flex: '0 1 20%',
+    margin: '1%',
+    display: 'flex',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+    textDecoration: 'none',
+  };
+  if (!politician) return <div />;
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: how about `return null` ?

Keeps us from rendering empty divs

File: src/Components/Politicians/PoliticianCard/index.js [hunk 5]
```diff
@@ -1,18 +1,35 @@
...
   return (
-    <Card style={cardStyle}>
-      <CardHeader title={name} avatar="http://lorempixel.com/200/200/" />
-    </Card>
+    <Link style={containerStyle} to={`/politicians/${politician.id}`}>
+      <Paper style={{ flex: '1 1 100%', display: 'flex', flexWrap: 'wrap' }}>
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: Could we keep styles outside of the JSX?

File: src/Components/Politicians/PoliticianCard/index.js [hunk 6]
```diff
@@ -2,14 +2,16 @@
...
 import type { PoliticianType } from '../../../types';
 
 export type Props = {
   politician: PoliticianType,
+  progress: number,
 };
 
 const PoliticianCard = (props: Props) => {
-  const { politician } = props;
+  const { politician, progress } = props;
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: 👏

File: src/Components/Politicians/PoliticianCard/index.js [hunk 7]
```diff
@@ -28,6 +30,14 @@ const PoliticianCard = (props: Props) => {
           style={{ flex: '1 1 100%' }}
         />
         <p style={{ flex: '1 1 100%', textAlign: 'center' }}>{politician.name}</p>
+        <span style={{ marginLeft: `${progress}%` }}>
+          {progress}%
+        </span>
+        <LinearProgress
+          style={{ flex: '1 1 100%' }}
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: Could you ELI5 what the flex properties do?

File: src/Components/Politicians/PoliticianCard/index.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ import Paper from 'material-ui/Paper';
 import { Link } from 'react-router-dom';
 import LinearProgress from 'material-ui/LinearProgress';
 import type { PoliticianType } from '../../../types';
+import * as styles from './styles';
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: gah saw this commit now :S

If you're interested, I can show you of the ways I rewrite git history.

To avoid exactly these kind of situations

File: src/Components/Politicians/PoliticianCard/styles.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+export const container = {
+  flex: '0 1 20%',
+  margin: '1%',
+  display: 'flex',
+  flexWrap: 'wrap',
+  alignItems: 'center',
+  textDecoration: 'none',
+};
+
+export const flexItem = {
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/styles.js]: I'm not too happy with the `flexItem` name here

Maybe a name documenting the purpose?

File: src/Components/Politicians/PoliticianCard/styles.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+  flex: '1 1 100%',
+};
+
+export const name = {
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/styles.js]: Hmm, first time I've seen an object spread in a style item.

Could you explain the reasoning behind it?
Rather than assigning two classes to the HTML element?

File: src/App.js [hunk 11]
```diff
@@ -7,6 +7,7 @@ import { BrowserRouter as Router, Route } from 'react-router-dom';
 
 import NavBar from './Components/NavBar';
 import Politicians from './Components/Politicians/Politicians';
+import PoliticianItemContainer from './Components/Politicians/PoliticianItemContainer';
```
Comment (Ragnar-H) [inline src/App.js]: Please say if Flow is being more of an hindrance than useful.
I'm always happy to help with typings :)

File: src/Components/Politicians/PoliticianCard/index.js [hunk 12]
```diff
@@ -23,10 +22,10 @@ const PoliticianCard = (props: Props) => {
           style={styles.flexItem}
         />
         <p style={styles.name}>{politician.name}</p>
-        <span style={{ marginLeft: `${progress}%` }}>
-          {progress}%
+        <span style={{ marginLeft: `${0}%` }}>
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianCard/index.js]: how about writing `const progress = 0` until this feature has been fixed?

It'd tell me more about the purpose than the magic number of 0

File: src/Components/Politicians/PoliticianItem/stories/PoliticianItem.stories.js [hunk 13]
```diff
@@ -3,12 +3,9 @@ import React from 'react';
...
 
 storiesOf('Politician Item', module)
   .addDecorator(muiTheme())
-  .add('High ranked politician', () =>
-    <PoliticianItem id={1} name="Jon Jonsson" progress={53} />
-  )
-  .add('Low ranked politician', () =>
-    <PoliticianItem id={1} name="Birgitta Haukdal" progress={23} />
-  );
+  .add('Politician 1', () => <PoliticianItem politician={politicians[0]} />)
```
Comment (Ragnar-H) [inline src/Components/Politicians/PoliticianItem/stories/PoliticianItem.stories.js]: Would you mind writing more descriptive story names?

File: src/types.js [hunk 14]
```diff
@@ -42,7 +50,7 @@ export type PoliticianType = {
   districtNumber: number,
   party: PartyId,
   district: DistrictType,
-  promises: Array<PromiseType>,
+  promises: Array<number>,
```
Comment (Ragnar-H) [inline src/types.js]: I've been trying to write `SomeTypeId` in these cases.

I realize it's more terse, but it instantly tells me what the data-structure I'm looking at is.

If you don't agree with that, then I'll stop doing it. No use in having a convention if not everybody follows it

File: src/utils/testFixtures.js [hunk 15]
```diff
@@ -1,4 +1,6 @@
 /* @flow */
+/* eslint-disable */
```
Comment (Ragnar-H) [inline src/utils/testFixtures.js]: do you need to disable all the rules?

Comment (AriHrannar) [reply src/utils/testFixtures.js]: yes otherwise it will complain over magic numbers and such (ids in arrays)

Comment (Ragnar-H) [reply src/utils/testFixtures.js]: how many rules is it complaining about?

If it's less than 5 I'd prefer it if you turned off just those rules

File: src/App.js (hunk 11 above)
Comment (AriHrannar) [reply src/App.js]: Id like to learn Flow so its not a hinderance now

File: src/utils/testFixtures.js (hunk 15 above)
Comment (AriHrannar) [reply src/utils/testFixtures.js]: its just magic numbers, I can just disable that rule and then we get to keep all the others. I think we should at least have those arrays of ids for testing

Comment (Ragnar-H) [reply src/utils/testFixtures.js]: Yes, I totally agree that we should have these tests.

I'm just reluctant to nuke all the lint rules at once
